### PR TITLE
Update package template

### PR DESCRIPTION
**Pull request by supechicken**
### Changes
- Merge with package template in [chromebrew/chromebrew.github.io](https://github.com/supechicken/chromebrew.github.io/blob/main/index.html)
- Add `#!/usr/bin/env ruby` to enable syntax highlighting

### CODE REVIEW

**Comment by satmandu** [comment]
Do we want to have `#!/usr/bin/env ruby` in every package file? Those files are set to 0644 on purpose so they aren't considered executable.

@uberhacker ?

**Comment by uberhacker** [comment]
> Do we want to have `#!/usr/bin/env ruby` in every package file? Those files are set to 0644 on purpose so they aren't considered executable.
> 
> @uberhacker ?

I generally prefer `#!/usr/local/bin/ruby` but that might mess up GitHub language detection.

File: packages/template.rb.template [hunk 1]
```diff
@@ -1,73 +1,69 @@
...
   #
 
-  # Function to perform pre-flight operations prior to dependency checks.
+  # Function for checks to see if install should occur.
   def self.preflight
 
   end
 
+  # Function to perform pre-install operations prior to install for both source build and binary distribution.
+  def self.preinstall
```
Comment (uberhacker) [inline packages/template.rb.template:42]: Already on line 66?

File: packages/template.rb.template [hunk 2]
```diff
@@ -80,4 +76,9 @@ class Template < Package
   end
 
 
-end
+  # Function to perform check from source build.
+  # This is execute if and only if `crew build`.
+  def self.check
```
Comment (uberhacker) [inline packages/template.rb.template:81]: I arranged these in chronological sequence.  I like to have the sequence in packages too but so many packagers like to change it around. :)

**Review by uberhacker — CHANGES_REQUESTED** [review]

File: packages/template.rb.template [hunk 3]
```diff
@@ -1,73 +1,69 @@
...
 
 class Template < Package
   description ''
   homepage ''
   version ''
-  license ''
-  compatibility ''
-  source_url '*.tar.gz'
+  license '' # Package license
+  compatibility '' # Package compatibility
```
Comment (uberhacker) [inline packages/template.rb.template:12]: Maybe show some examples too?
```
license '' # Package license (eg. GPL-3, MIT, etc.)
compatibility '' # Package compatibility (eg. aarch64, armv7l, i686, x86_64 or all)
```

File: packages/template.rb.template [hunk 4]
```diff
@@ -1,73 +1,69 @@
...
+  source_url '*.tar.{gz,bz2,lz,etc}'
```
Comment (uberhacker) [inline packages/template.rb.template:13]: How about?
```
source_url '' # The url to the source code (eg. https://ftpmirror.gnu.org/sed/sed-4.8.tar.xz)
```

**Review by uberhacker — CHANGES_REQUESTED** [review]

**Comment by uberhacker** [comment]
> > Do we want to have `#!/usr/bin/env ruby` in every package file? Those files are set to 0644 on purpose so they aren't considered executable.
> > @uberhacker ?
> 
> I generally prefer `#!/usr/local/bin/ruby` but that might mess up GitHub language detection.

Thinking outside the box.  Why do we care about language detection?  Maybe it would be less confusing if we don't add the line?

**Comment by supechicken** [comment]
> Thinking outside the box. Why do we care about language detection? Maybe it would be less confusing if we don't add the line?

Because I prefer syntax highlighting :)

**Comment by uberhacker** [comment]
> > Thinking outside the box. Why do we care about language detection? Maybe it would be less confusing if we don't add the line?
> 
> Because I prefer syntax highlighting :)

Isn't that controlled in your editor?

File: packages/template.rb.template (hunk 4 above)
Comment (supechicken) [reply packages/template.rb.template]: Added

File: packages/template.rb.template (hunk 2 above)
Comment (supechicken) [reply packages/template.rb.template]: Reverted to the original sequence

**Comment by supechicken** [comment]
> Can you mention in the comment that the `#!/usr/bin/env ruby` is only for the template file, and that it does not go into a normal package file?

Now the shebang line looks like this:
```ruby
#!/usr/bin/env ruby <- This line is not necessary, just for GitHub's language detection
```

File: packages/template.rb.template [hunk 5]
```diff
@@ -1,73 +1,69 @@
-# It is recommended that you investigate other files in the package directory
-# to help get an idea of how existing packages are being installed through chromebrew
+#!/usr/bin/env ruby <- This line is not necessary, just for GitHub's language detection
```
Comment (uberhacker) [inline packages/template.rb.template:1]: Do we need another `#` before `<-` to indicate the rest of the line is a comment?

File: packages/template.rb.template [hunk 6]
```diff
@@ -1,73 +1,69 @@
...
-
   # Function to perform build from source.
   def self.build
 
   end
 
-
   # Function to perform check from source build.
-  # This executes only during `crew build`.
+  # This is execute if and only if `crew build`.
```
Comment (uberhacker) [inline packages/template.rb.template:57]: `This is executed ...`?  Proper verb usage. :)

File: packages/template.rb.template (hunk 5 above)
Comment (supechicken) [reply packages/template.rb.template:1]: The shebang line won't work anyways as `env` does not support comments

File: packages/template.rb.template [hunk 7]
```diff
@@ -1,73 +1,69 @@
...
+  #  })
 
 
   # Register dependencies (use the following line as a basis)
   #
   # depends_on '*'
   #
 
-  # Function to perform pre-flight operations prior to dependency checks.
+  # Function for checks to see if install should occur.
```
Comment (uberhacker) [inline packages/template.rb.template:36]: How about?
```
# Function to check if install should occur.
```

File: packages/template.rb.template (hunk 6 above)
Comment (supechicken) [reply packages/template.rb.template]: Change made

File: packages/template.rb.template [hunk 8]
```diff
@@ -1,73 +1,69 @@
...
-
   # Function to perform build from source.
   def self.build
 
   end
 
-
   # Function to perform check from source build.
-  # This executes only during `crew build`.
+  # This is executed if and only if `crew build`.
```
Comment (uberhacker) [inline packages/template.rb.template:57]: How about?
```
# This is executed only during `crew build`.
```

File: packages/template.rb.template (hunk 7 above)
Comment (supechicken) [reply packages/template.rb.template]: Change made

**Comment by supechicken** [comment]
Please also take a look at https://github.com/chromebrew/chromebrew.github.io/pull/2 :)

**Review by uberhacker — APPROVED** [review]

**Pull request merged**